src/main/java/seedu/address/model/listing/Listing.java

```diff
@@ -0,0 +1,107 @@
+package seedu.address.model.listing;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.ArrayList;
+import java.util.Objects;
+import java.util.UUID;
+
+/**
+ * Represents a Listing in GoodMatch.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Listing {
+    // Compulsory fields
+    private Title title;
+    private Description description;
+    private String uniqueID = UUID.randomUUID().toString();
+
+    // Optional fields
+    private ArrayList<String> applicants = new ArrayList<>(); // TODO: Update when Applicant class is added
+
+    /**
+     * Every field must be present and not null.
+     */
+    public Listing(Title title, Description description, ArrayList<String> applicants) {
+        requireAllNonNull(title, description, applicants);
+        this.title = title;
+        this.description = description;
+        this.applicants.addAll(applicants);
+    }
+
+    public Title getTitle() {
+        return title;
+    }
+
+    public Description getDescription() {
+        return description;
+    }
+
+    public ArrayList<String> getApplicants() {
+        return applicants;
+    }
+
+    public String getUniqueID() {
+        return uniqueID;
+    }
+
+    /**
+     * Returns true if both listings have the same title.
+     * This defines a weaker notion of equality between two listings.
+     */
+    public boolean isSameListing(Listing otherListing) {
+        if (otherListing == this) {
+            return true;
+        }
+
+        return otherListing != null
+                && otherListing.getTitle().equals(getTitle());
+    }
+
+    /**
+     * Returns true if both persons have the same identity and data fields.
+     * This defines a stronger notion of equality between two persons.
```

Comment:
Change comment to listings instead of persons

Reply:
updated